src/dialects/oracle/connection-manager.js

```diff
@@ -0,0 +1,192 @@
+
+      if (config.port && config.port !== 0) {
+        connectString += `:${config.port}`;
+      } else {
+        connectString += ':1521'; //Default port number
+      }
+      connectString += `/${config.database}`;
+      config.database = connectString;
+    }
+  }
+
+  // Expose this as a method so that the parsing may be updated when the user has added additional, custom types
+  _refreshTypeParser(dataType) {
+    parserStore.refresh(dataType);
+  }
+
+  _clearTypeParser() {
+    parserStore.clear();
+  }
+
+  /**
+   * Connect with Oracle database based on config, Handle any errors in connection
+   * Set the pool handlers on connection.error
+   * Also set proper timezone once connection is connected.
+   *
+   * @param {object} config
+   * @returns {Promise<Connection>}
+   * @private
+   */
+  async connect(config) {
+    const connectionConfig = {
+      user: config.username,
+      host: config.host,
+      port: config.port,
+      database: config.database,
+      password: config.password,
+      externalAuth: config.externalAuth,
+      stmtCacheSize: 0,
+      connectString: config.database,
+      ...config.dialectOptions
+    };
+
+    try {
+      // Check the config object
+      this.checkConfigObject(connectionConfig);
+
+      // We assume that the database has been correctly formed
+      connectionConfig.connectString = connectionConfig.database;
+
+      // We check if there are dialect options
+      if (config.dialectOptions) {
+        // const dialectOptions = config.dialectOptions;
+
+        // //If stmtCacheSize is defined, we set it
+        // if (dialectOptions && 'stmtCacheSize' in dialectOptions) {
+        //   connectionConfig.stmtCacheSize = dialectOptions.stmtCacheSize;
+        // }
+
+        Object.keys(config.dialectOptions).forEach(key => {
+          connectionConfig[key] = config.dialectOptions[key];
+        });
+      }
+
+      const connection = await this.lib.getConnection(connectionConfig);
+
+      debug('connection acquired');
+      connection.on('error', error => {
+        switch (error.code) {
+          case 'ESOCKET':
+          case 'ECONNRESET':
+          case 'EPIPE':
+          case 'PROTOCOL_CONNECTION_LOST':
+            this.pool.destroy(connection);
+        }
+      });
+
+      return connection;
+    } catch (err) {
+      // We split to get the error number; it comes as ORA-XXXXX:
+      let errorCode = err.message.split(':');
+      errorCode = errorCode[0];
+
+      switch (errorCode) {
+        case 'ORA-28000': // Account locked
+        case 'ORA-12541': // ORA-12541: TNS:No listener
+          throw new SequelizeErrors.ConnectionRefusedError(err);
+        case 'ORA-01017': // ORA-01017 : invalid username/password; logon denied
+          throw new SequelizeErrors.AccessDeniedError(err);
+        case 'ORA-12154':
+          throw new SequelizeErrors.HostNotReachableError(err); //ORA-12154: TNS:could not resolve the connect identifier specified
```

Comment:
Missing whitespace after the '//' :)

But more importantly, there are a huge number of errors number that could/should be added here. Some additional common errors:

ORA-12170 TNS: Connect Timeout occurred 
ORA-12505 TNS:listener does not currently know of SID given in connect descriptor
ORA-12560 TNS: Protocol Adapter Error 
ORA-28040 No matching authentication protocol

Reply:
Added all the ORA errors mentioned above.